src/ocean/meta/traits/Basic.d

```diff
@@ -0,0 +1,508 @@
+
+public template isComplexType( T )
+{
+    const bool isComplexType =
+           is( Unqual!(T) == cfloat )
+        || is( Unqual!(T) == cdouble )
+        || is( Unqual!(T) == creal );
+}
+
+///
+unittest
+{
+    static assert ( isComplexType!(cdouble));
+    static assert ( isComplexType!(Const!(cdouble)));
+    static assert (!isComplexType!(double));
+}
+
+/*******************************************************************************
+
+    Params:
+        T = type to check
+
+    Returns:
+        `true` if `T` is a built-in imaginary floating point type
+
+*******************************************************************************/
+
+public template isImaginaryType( T )
+{
+    const bool isImaginaryType =
+           is( Unqual!(T) == ifloat )
+        || is( Unqual!(T) == idouble )
+        || is( Unqual!(T) == ireal );
+}
+
+///
+unittest
+{
+    static assert ( isImaginaryType!(idouble));
+    static assert ( isImaginaryType!(Const!(idouble)));
+    static assert (!isImaginaryType!(double));
+}
+
+/*******************************************************************************
+
+    Params:
+        T = type to check
+
+    Returns:
+        `true` if `T` is any built-in floating-point type: real, complex, or
+        imaginary.
+
+*******************************************************************************/
+
+public template isFloatingPointType( T )
+{
+    const bool isFloatingPointType =
+           isRealType!(T)
+        || isComplexType!(T)
+        || isImaginaryType!(T);
+}
+
+///
+unittest
+{
+    static assert (isFloatingPointType!(double));
+    static assert (isFloatingPointType!(ifloat));
+}
+
+/*******************************************************************************
+
+    Used by other traits to distinguish between dynamic and static arrays
+    instead of plain `bool` values.
+
+    `ArrayKind.NotArray` is explicitly defined to have value `0` so that it can
+    be in regular condition, i.e. `static if (isArrayType!(T))`.
+
+*******************************************************************************/
+
+public enum ArrayKind
+{
+    NotArray    = 0,
+    Static      = 1,
+    Dynamic     = 2,
+    Associative = 3,
+}
+
+/*******************************************************************************
+
+    Check if type is an array type and which kind of array it is
+
+    Params:
+        T = type to check
+
+    Returns:
+        `ArrayKind` value indicating if `T` is an array and if it is, static or
+        dynamic
+
+*******************************************************************************/
+
+public template isArrayType ( T )
+{
+    static if (is(Unqual!(T) U == U[]))
+        const isArrayType = ArrayKind.Dynamic;
+    else static if (is(Unqual!(T) U : U[]))
+        const isArrayType = ArrayKind.Static;
+    else static if (is(typeof(T.init.values[0])[typeof(T.init.keys[0])] ==
+            Unqual!(T)))
+        const isArrayType = ArrayKind.Associative;
```

Comment:
The syntax you link can only be used as part of template declaration in D1 and not in `is` expression. I don't want to add define new templates just to make a single line that no one will ever look at prettier ;)